packages/core/contracts/merkle-distributor/implementation/MerkleDistributor.sol

```diff
@@ -30,10 +30,12 @@ contract MerkleDistributor is Ownable, Testable {
     struct Window {
         // Claims for this window can begin after `start`.
         uint256 start;
-        // Merkle root describing the distribution.
-        bytes32 merkleRoot;
         // Currency in which reward is processed.
         IERC20 rewardToken;
+        // Merkle root describing the distribution.
+        bytes32 merkleRoot;
+        // IPFS hash of the merkle tree. Can by used to independently fetch recipient proofs.
+        string ipfsHash;
```

Comment:
I'm probably missing this in an obvious place but where does the value come from adding `ipfsHash` to the onchain struct?

Reply:
It is produced by the IPFS upload step; this returns the IPFS hash. Take a look in the `2_PublishClaimsForWindow.ts`. In this script it is the last param in the `setWindow` function call.